### PR TITLE
[metrics] change default behaviour of state dict

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes
https://github.com/PyTorchLightning/pytorch-lightning/issues/4361
https://github.com/PyTorchLightning/pytorch-lightning/issues/4645
https://github.com/PyTorchLightning/pytorch-lightning/issues/4666

As default metrics add their internal state to the models `state_dict`. However, it seems counter intuitive to users (already 3 issues on this specific topic), and even though we since v1.0.6 provides a `.persistent(mode)` to change this behavior, the feedback indicates that it is probably better as default to not add the metric state to `state_dict` (and then the method can be used to switch it on). 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by teddykoker — APPROVED** [review]
Makes sense to me, I think I overlooked this when we first implemented them

File: docs/source/metrics.rst [hunk 1]
```diff
@@ -133,8 +133,8 @@ This metrics API is independent of PyTorch Lightning. Metrics can directly be us
 
 .. note::
 
-    Metric states will as default add their internal state to the models ``state_dict``.
-    To change this after initializing the metric the method ``.persistent(mode)`` can
+    Metric states are **not** as default added to the models ``state_dict``.
+    To change this after initializing the metric, the method ``.persistent(mode)`` can
```
Comment (rohitgr7) [inline docs/source/metrics.rst:137]:
```suggestion
    Metric states are **not** added to the models ``state_dict`` by default.
    To change this, after initializing the metric, the method ``.persistent(mode)`` can
```

**Review by rohitgr7 — APPROVED** [review]
LGTM. Just one comment.

**Review by tchaton — APPROVED** [review]
LGTM !
